### ISSUE TITLE
downloadProgressCallback not working in IOS

### ISSUE THREAD

**Issue by ahsanparacha02** [microsoft/react-native-code-push#2790]
When upgrading from version v8.0.2 to v9.0.1 in React Native 0.76.5, the downloadProgressCallback does not trigger for iOS, while it works fine on Android. It only triggers once the bundle download is complete.

Here is code snippet.

`
async sync() {
    CodePush.sync(
      { installMode: CodePush.InstallMode.IMMEDIATE, updateDialog: false },
      this.codePushStatusDidChange,
      this.codePushDownloadDidProgress
    );
    CodePush.allowRestart();
  }

  codePushDownloadDidProgress(event){
    this.setState({
      progress: (event.receivedBytes / event.totalBytes).toFixed(2) * 100,
    });
    MyToast.popToast(
      `Progress ${(event.receivedBytes / event.totalBytes).toFixed(2) * 100}`
    );
  };
`

**Comment by ahsanparacha02** [comment]
> Hi [@ahsanparacha02](https://github.com/ahsanparacha02), Thank you for reporting this issue. Could you please provide us demo app so we could reproduce this issue?

**Android**:

![Image](https://github.com/user-attachments/assets/483511af-cba0-446f-b084-28036f8cb53d)

**Iphone**

![Image](https://github.com/user-attachments/assets/48b9c31a-f575-4c30-828c-fd3c273b5ef4)

Mentioned screenshot shows that the iOS progress bar is not updating. The receivedBytes and totalBytes events are not being triggered on iOS.

`codePushDownloadDidProgress(event){
this.setState({
progress: (event.receivedBytes / event.totalBytes).toFixed(2) * 100,
});};`

**Comment by IlyaBausovAkvelon** [comment]
@ahsanparacha02 Could you please provide demo application so we could use it to reproduce issue on our side?

**Comment by ahsanparacha02** [comment]
> [@ahsanparacha02](https://github.com/ahsanparacha02) Could you please provide demo application so we could use it to reproduce issue on our side?

Do you need ipa file?

**Comment by felipegmg** [comment]
same problem!

**Comment by ahsanparacha02** [comment]
> We need a small demo project in which you can reproduce this issue, that way we can debug the source code and see what's going on. Ipa won't do it since it is an archive containing compiled code.

https://github.com/ahsanparacha02/RNCodepush.git

![Image](https://github.com/user-attachments/assets/da7a766e-9d0b-4052-abb7-791c83d2692e)

**Comment by MikhailSuendukov** [comment]
Hello, I tried to run your app but when I run npm run android I get the following error:

> % npm run android
> 
> > RNCodepush@0.0.1 android
> > react-native run-android
> 
> (node:19313) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
> (Use `node --trace-deprecation ...` to show where the warning was created)
> info Installing the app...
> 
> info 💡 Tip: Make sure that you have set up your development environment correctly, by running npx react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor
> 
> error Failed to install the app. Command failed with Unknown system error -8: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
> spawn Unknown system error -8.
> info Run CLI with --verbose flag for more details.

And this error does not occur to me on a newly created project, so I think that the issue may lie in the demo project. Сould you please help me with solving this problem and provide steps by which I can reproduce your original error.

**Comment by ahsanparacha02** [comment]
Can u send me the steps you perform to run application

**Comment by ahsanparacha02** [comment]
Steps
- node >18
- yarn install
- sync project in android studio
- yarn android

Please follow these steps,and let me know in case any issue

Thanks

**Comment by ahsanparacha02** [comment]
Please pull latest code from repo before perform above steps

**Comment by ahsanparacha02** [comment]
> We need a small demo project in which you can reproduce this issue, that way we can debug the source code and see what's going on. Ipa won't do it since it is an archive containing compiled code.

any update ?

**Comment by MikhailSuendukov** [comment]
I added additional logging in `codePushDownloadDidProgress` and confirmed that it is being triggered. I successfully tested the iOS version and observed that the app immediately downloads the update, the progress bar jumps to 100%, and then the app restarts. After the restart, the label displays 0%. This behavior intentional by design.

**Comment by ahsanparacha02** [comment]
> I added additional logging in `codePushDownloadDidProgress` and confirmed that it is being triggered. I successfully tested the iOS version and observed that the app immediately downloads the update, the progress bar jumps to 100%, and then the app restarts. After the restart, the label displays 0%. This behavior intentional by design.

Yes, you are correct. However, if the bundle size increases above 10MB, the issue will replicate. Right now, in my app, the progress bar initially shows 0%. Once the bundle download is complete, it jumps directly to 100%. The progress between 0% and 100% is not being triggered.

**Comment by ahsanparacha02** [comment]
@MikhailSuendukov any update ?

**Comment by ahsanparacha02** [comment]
> We need a small demo project in which you can reproduce this issue, that way we can debug the source code and see what's going on. Ipa won't do it since it is an archive containing compiled code.

**Comment by ahsanparacha02** [comment]
@IlyaBausovAkvelon @DordeDimitrijev any update ?

**Comment by IlyaBausovAkvelon** [comment]
Hi @ahsanparacha02,
We were able to reproduce the issue. Unfortunately, we are not planning to fix this issue due to App Center deprecation.

**Comment by ahsanparacha02** [comment]
> Hi [@ahsanparacha02](https://github.com/ahsanparacha02), We were able to reproduce the issue. Unfortunately, we are not planning to fix this issue due to App Center deprecation.

i have migrated app center to code push server, but still getting same issue

**Comment by IlyaBausovAkvelon** [comment]
@ahsanparacha02,
The issue is in react-native-code-push iOS implementation. It does not related to CodePush backend.

**Comment by ahsanparacha02** [comment]
> [@ahsanparacha02](https://github.com/ahsanparacha02), The issue is in react-native-code-push iOS implementation. It does not related to CodePush backend.

Are you going to fixed this issue for IOS ?

**Comment by DmitriyKirakosyan** [comment]
As @IlyaBausovAkvelon mentioned, we do not plan to fix this issue. Closing.
